# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 1e72f84b4206
# Full Hash: 1e72f84b420612e65ef17b820faeb1b6ad41e4b0
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 03:48:30
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r 25281e1775c9 -r 1e72f84b4206 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:38 2020 +0000
@@ -4410,12 +4410,11 @@
   explicit WasmABIArgIter(const VecT& types)
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
-
 }  // namespace jit
 
 namespace wasm {
-TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp);
-TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp);
+const TlsData* ExtractCalleeTlsFromFrameWithTls(const Frame* fp);
+const TlsData* ExtractCallerTlsFromFrameWithTls(const Frame* fp);
 }  // namespace wasm
 
 }  // namespace js